api.go

```diff
@@ -163,6 +163,10 @@ type ChunkMatch struct {
 	// its length will equal that of Ranges. Any of its elements may be nil.
 	SymbolInfo []*Symbol
 
+	// CaseSensitiveMatch tracks whether the pattern and match have
+	// the same case. Its length equals that of Ranges.
+	CaseSensitiveMatch []bool
```

Comment:
Feels weird this makes it into our API. I wonder if we can do something more generic to smuggle this along? eg a little booster score or something?

Reply:
We need to somehow transport that information from where we match to where we score. We don't have to export the field, but isn't it similar to `SymbolInfo` in the sense that it carries meta-information about the match?